go/libraries/doltcore/row/row.go

```diff
@@ -114,20 +114,51 @@ func GetFieldByNameWithDefault(colName string, defVal types.Value, r Row, sch sc
 	}
 }
 
-// ReduceToIndex creates an index record from a primary storage record.
-func ReduceToIndex(idx schema.Index, r Row) (Row, error) {
-	newRow := nomsRow{
-		key:   make(TaggedValues),
-		value: make(TaggedValues),
-		nbf:   r.Format(),
-	}
+// ReduceToIndexKeys creates a full key and a partial key from the given row (first tuple being the full key). As all
+// non-key tuples in an index map are empty, the full key is equivalent to the entire row of an index. The partial key
+// contains only the stated indexed columns, allowing for matches across duplicate values (for non-unique indexes).
+func ReduceToIndexKeys(idx schema.Index, r Row) (types.Tuple, types.Tuple, error) {
```

Comment:
I don't get the "partial key" abstraction, probably because I haven't been familiar with the index implementation code for a while now. But I can say this is a confusing abstraction. I don't know what a partial key is after reading this description, or why I would want it.

Does this need to be a single method? Maybe two methods, one for unique keys and one for non-unique, would make this a simpler read.